targets/TARGET_Silicon_Labs/TARGET_EFM32/TARGET_EFM32GG/device/TARGET_1024K/TOOLCHAIN_ARM_MICRO/efm32gg.sct

```diff
@@ -40,11 +42,11 @@ LR_IROM1  MBED_APP_START  MBED_APP_SIZE  {    ; load region size_region
    .ANY (+RO)
   }
 
-  RW_IRAM1  (MBED_RAM_START+VECTOR_SIZE)  (MBED_RAM_SIZE-VECTOR_SIZE)  {  ; RW data
+  RW_IRAM1  MBED_RAM1_START  (MBED_RAM_SIZE-VECTOR_SIZE)  {  ; RW data
    .ANY (+RW +ZI)
   }
 
-  ARM_LIB_HEAP  AlignExpr(+0, 16)  EMPTY  (MBED_RAM_SIZE-RAM_FIXED_SIZE+MBED_RAM_START-AlignExpr(ImageLimit(RW_IRAM1), 16) - MBED_CONF_TARGET_BOOT_STACK_SIZE)  {
+  ARM_LIB_HEAP  AlignExpr(+0, 16)  EMPTY  (MBED_RAM_SIZE-RAM_FIXED_SIZE+MBED_RAM1_START-AlignExpr(ImageLimit(RW_IRAM1), 16))  {
```

Comment:
This is convoluted to the point I can't really visualise it (and I think that's how the error wasn't spotted in the first place).

I think it's correct now - "heap size = total RAM size - fixed size - IRAM1 used space", with the used space being "AlignExpr(ImageLimit(RW_IRAM1, 16)) - MBED_RAM1_START", but the sign flippery confuses that.

Last time I touched this on a branch, I went with

    (MBED_RAM_START+MBED_RAM_SIZE-MBED_BOOT_STACK_SIZE)-AlignExpr(ImageLimit(RW_IRAM1), 16)

or "heap size = Stack start address - IRAM1 end address".

Actually, we can just say that more simply:

    ImageBase(ARM_LIB_STACK) - AlignExpr(ImageLimit(RW_IRAM1), 16)

although I guess you'd have to reorder the sections to decare stack first. Kind of makes sense though - you're laying out the fixed things and finally heap is what's left.

Or the original size-based formula could be somewhat simplified the same way:

    MBED_RAM_SIZE-RAM_FIXED_SIZE-AlignExpr(ImageLength(RW_IRAM1), 16)

But I think it's clearest to denote the heap by its direct boundaries (end of IRAM1, start of stack).

It would help if you could just set the limit as an absolute address rather than a size, then we could just say `ImageBase(ARM_LIB_STACK)` for that.

Reply:
Completely agree. `ImageLength()` looks useful, so we don't need to do _image start_ - `ImageLimit()`.

Also, some of the terms we put in `RAM_FIXED_SIZE`'s definition eventually get cancelled out in the maths of the heap size. But this PR is to make as small change as possible, so keeping the existing logic is fine.